conan/tools/files/files.py

```diff
@@ -185,7 +185,7 @@ def download(conanfile, url, filename, verify=True, retry=None, retry_wait=None,
     :param headers: A dictionary with additional headers
     :param md5: MD5 hash code to check the downloaded file
     :param sha1: SHA-1 hash code to check the downloaded file
-    :param sha256: SHA-256 hash code to check the downloaded file
+    :param sha256: SHA-256 hash code to check the downloaded file. If set, the download is cached.
```

Comment:
But this is not generally true. This only happens if the download cache is activated in the ``conf``, but that is not a default. On the other hand, if the cache is activated and the download does not define ``sha256``, a warning will be printed to the output, so it will be clear what is happening.